### ISSUE TITLE
Adding a Deye inverter with LSW-3 logger

### ISSUE THREAD

**Issue by akmadov** [AdyRock/ady.solarman#14]
Hi! For some reason I cannot find where and how to add my Deye inverter in the Sofar/Solarman Homey app. When I go through the maintanance option, the Homey app only requests for credentials and the only credentials accepted are my Homey ones. Where and how do I need to enter the credentials for logging into the Solarman APP or website, so that I can see values in the Homey app, other than zeroes. Thanks!

**Comment by AdyRock** [comment]
When you added the device, I presume you entered your Solarman credentials then?
Has it worked before and stopped working? just wondering because you mentioned the repair option.

**Comment by akmadov** [comment]
Thank you for the swift reply! I have just now started my attempt at using the Homey app. When I am adding a device it seems as if the credentials are filled "automatically" because I can see the field asking for username and password for a second but then the app jumps automatically to the next screen (with the device being "successfully" added). Just right now I opened Homey and I saw a popup saying "Rate limit". My attempt at trying the repair option is driven by reading through the forums.

**Comment by akmadov** [comment]
And now when adding a new device I see "Failed to get station list: appid insufficient allowance". Maybe I need to mentioned that currently I am not at home and I am not on the same network as my inverter. My previous attempts were made at home and that is where the device was "automatically" found and added but it was showing only zeroes, without ever asking me for any credentials.

**Comment by akmadov** [comment]
Dear Sir, I have tried adding the device again which was successful this time. My problem, however, is that my plant is registered under the Chinese data servers (I do not know why my installers choose to do this, but nevermind). This means that for me to use the API, I need to connect to api.solarmanpv.com, instead of the globalapi.solarmapv.com, also used by your app. It is not an issue for me, as it turns out that Homey will not do the trick for my plan, but thought I would just let you know in case you encounter a similar issue in the future. When entering the credentials for the chinese servers, the app returns the invalid username error, as expected. I created a dummy account on the global servers just to test whether the Homey app would work.
For context, I wanted to automate the turning on and off of my water heater when there is excess solar production which is something I deduce from the power with which the battery is being charged, i.e. when charging power is >2700W, the heater turns on via a Tuya wifi fuse. I cannot use a power diverter, unfortunately. I am doing the automation with Siri shortcuts directly with "Get contents of URL" from the API.
All the best!

**Comment by AdyRock** [comment]
OK, thanks for the information. Although the email I got from Solarman said that all accounts should be migrated across to the new global data server. The app did use api.solarmanpv.com until a few months back when they introduced the new global server and I had to change the app.

**Comment by akmadov** [comment]
My inverter is Deye and the installers used the Deye app to create my account ,which still has the option to choose China servers, I guess that is why my case is as it is.

**Comment by AdyRock** [comment]
Fair enough. At the moment it looks like I might have to remove the Solarman support from the app as I have recently found out there is a limit on the total number of API calls.
